structure/linkedlist/doubly.go

```diff
@@ -91,6 +91,69 @@ func (ll *Doubly[T]) DelAtEnd() (T, bool) {
 	return retval, true
 }
 
+// DelInMiddle deletes the node in middle of linked list and returns its value.
+// Returns false if list is empty
+func (ll *Doubly[T]) DelInMiddle() (T, bool) {
+	if ll.Head == nil {
+		var r T
+		return r, false
+	}
+
+	mid := 0
+	length := ll.Count()
+	mid = int(length / 2)
+
+	var prev *Node[T]
+	cur := ll.Head
+	count := 0
+	for count < mid {
+		prev = cur
+		cur = cur.Next
+		count++
+	}
+
+	prev.Next = cur.Next
+	return cur.Val, true
+}
```

Comment:
Do you really need this? Its clear that you can simply call:

```suggestion
func (ll *Doubly[T]) DelInMiddle() (T, bool) {
 	return ll.DelByPos[T](ll.Count/2)
 }
```